core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1348,6 +1348,28 @@ public static void writeOperationNoSize(BufferedChecksumStreamOutput out, Transl
     }
 
     /**
+     * Gets the minimum generation that could contain the sequence number, or the current generation if there is no generation with the
+     * specified sequence number between the minimum and maximum sequence numbers.
+     *
+     * @param seqNo the sequence number
+     * @return the minimum generation for the sequence number, or the current generation
+     */
+    public TranslogGeneration getMinGenerationForSeqNo(final long seqNo) {
+        try (ReleasableLock ignored = writeLock.acquire()) {
+            final long minTranslogFileGeneration = readers
+                    .stream()
+                    .filter(r -> {
+                        final Checkpoint checkpoint = r.getCheckpoint();
+                        return checkpoint.minSeqNo <= seqNo && seqNo <= checkpoint.maxSeqNo;
```

Comment:
hmm... why do we need to check the lower bound? isn't it simpler to check `seqNo <= checkpoint.maxSeqNo` alone? I read the logic to be "min generation that has ops that are required if we recover from seqNo onwards", which maps to "has ops about this point"